src/main/java/org/terracotta/auditor/verifier/EndTimestampComparator.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.auditor.verifier;
+
+import java.util.Comparator;
+
+public class EndTimestampComparator implements Comparator<Operation> {
+  @Override
+  public int compare(Operation op1, Operation op2) {
+    long endTS1 = op1.getEndTS();
+    long endTS2 = op2.getEndTS();
+
+    return Long.signum(endTS1 - endTS2);
```

Comment:
Better replace that with ```Long.compare(endTS1, endTS2)```

Reply:
It doesn't do the same thing. You aren't supposed to directly compare two System.nanoTime() results - only their difference.

Reply:
Better to replace it all with `Comparators.comparingLong(Operation::getEndTS)`

Reply:
It's not that timestamps can overflow because lots of time passes, but just that the starting time point isn't well defined.

When writing the code, I already considered and rejected both of the suggestions made.

Why would I change the code so that it has a bug?

Reply:
Okay so we push out the overflow point as far in to the future as possible by doing the comparison in this way.  This is going to need a comment though to prevent it getting a accidentally changed in the future.  I'd also switch to using a lambda expression to avoid having a whole class for this (it's only used in the OrderDeterminer right now right?).

Reply:
With this in mind... is this the most accurate form then: `return Long.signum(Math.subtractExact(endTS1, endTS2));`?

Reply:
Please also keep in mind that there are many other places that do that comparison wrong, and assume that the timestamp will never overflow.

What we probably should do is to create a TimestampUtil class with a helper static method containing the comparison code. This could be explicitly documented, and that would get rid of ambiguities and moot questioning in the future.